code/main/edge/stateStoreManager.brs

```diff
@@ -38,8 +45,92 @@ function _adb_StateStoreManager() as object
                 return
             end if
 
-            m.setStateStore(handle.payload)
+            for each payload in handle.payload
+                m._addToStateStore(payload)
+            end for
+
         end function,
+
+        _addToStateStore: function(payload as object) as void
+            if _adb_isEmptyOrInvalidMap(payload)
+                _adb_logDebug("_adb_StateStoreManager::setStateStore() - stateStore payload is empty or invalid.")
+                return
+            end if
+
+            if _adb_isEmptyOrInvalidString(payload.key)
+                _adb_logDebug("_adb_StateStoreManager::setStateStore() - payload key is empty or invalid.")
+                return
+            end if
+
+            m._states[payload.key] = _adb_StateStore(payload)
+            _adb_logDebug("_adb_StateStoreManager::setStateStore() - stateStore updated to: (" + FormatJson(m._states) + ").")
+
+        end function,
+
+        _deleteStateStore: function(stateNames as object) as void
+            for each state in stateNames
+                m._states.delete(state)
+                _adb_logVerbose("_adb_StateStoreManager::_deleteStateStore() - stateStore with key:(" + FormatJson(state) + ") deleted.")
+            end for
+
+            _adb_logVerbose("_adb_StateStoreManager::_deleteStateStore() - stateStore updated to: (" + FormatJson(m._states) + ").")
+        end function
     }
 
 end function
+
+function _adb_StateStore(payload as object) as object
+
+    stateStore = {
+        _payload: payload,
+        _expiryTSInMillis: invalid,
+        _initTSInMillis: invalid,
+
+        init: function()
+            _adb_logVerbose("_adb_StateStore::init() - Intializing stateStore with payload: (" + FormatJson(m._payload) + ").")
+
+            if _adb_isEmptyOrInvalidMap(m._payload)
+                _adb_logDebug("_adb_StateStore::init() - stateStore payload is empty or invalid.")
+                return invalid
+            end if
+
+            maxAge = m._payload.maxAge
+            if _adb_isInvalidInt(maxAge)
+                _adb_logDebug("_adb_StateStore::init() - payload.maxAge is invalid, using 0 as default value.")
+                maxAge = 0
+            end if
+
+            m._initTSInMillis = _adb_timestampInMillis()
+            m._expiryTSInMillis = m._initTSInMillis + (maxAge * 1000)
```

Comment:
We have the same logic in`locationHintManager`. Can we extract it into a utility function?

Reply:
Just added.